src/jit/codegenarm64.cpp

```diff
@@ -2712,9 +2712,8 @@ void CodeGen::genLockedInstructions(GenTreeOp* treeNode)
     genDefineTempLabel(labelRetry);
 
     emitAttr dataSize = emitActualTypeSize(data);
+
```

Comment:
Would be good to update the comment above, which shows the code that would be generated

Reply:
Just a comment for future, not suggesting for this change - from #17508, it looks like for an Exchange where the result is not used, it can just be `stlr, dmb`. Is info about whether the result is used available here?

Reply:
> update the comments

Done

Reply:
> I would rather not let the JIT degenerate the Exchange() to a Volatile.Write(). In some senses it is like an unused volatile read that shouldn't be optimized away per the C# spec.

True, a volatile read is eliminated but there are greater restrictions in place than a volatile read requires so I think it's doable, up to you

Reply:
It may be doable, my gut reaction is/was contaminated by years in the embedded world where volatile reads may pull an item from a fifo or clear an error....

I'll keep the potential optimization in mind.

Reply:
Makes sense, thanks